etna/analysis/plotters.py

```diff
@@ -186,3 +187,62 @@ def plot_anomalies(
         ax[i].scatter(anomaly, segment_df[segment_df.index.isin(anomaly)]["target"].values, c="r")
 
         ax[i].tick_params("x", rotation=45)
+
+
+def get_correlation_matrix(
+    ts: TSDataset, segments: Optional[List[str]] = None, method: Optional[str] = "pearson"
+) -> np.array:
+    """Compute pairwise correlation of timeseries for selected segments.
+
+    Parameters
+    -----------
+    ts :
+        TSDataset with timeseries data
+    segments:
+        Segments to use
+    method:
+        Method of correlation:
+        pearson : standard correlation coefficient
+        kendall : Kendall Tau correlation coefficient
+        spearman : Spearman rank correlation
+
+    Returns
+    -------
+    Correlation matrix
+    """
+    if method not in ["pearson", "kendall", "spearman"]:
+        raise ValueError(f"'{method}' is not a valid method of correlation.")
+    if segments is None:
+        segments = sorted(ts.segments)
+    correlation_matrix = ts[:, segments, "target"].corr(method=method).values
```

Comment:
can we use all the series, not only `target` to compute correlation?
it would be useful for `target` x `exog_features` correlation research

Reply:
Done